### PR TITLE
ci: use reusable workflows

### PR DESCRIPTION
**Pull request by arys-splunk**
Docs: https://docs.github.com/en/actions/learn-github-actions/reusing-workflows

This is already working in UCC-related open source libraries.

### CODE REVIEW

File: enforce/.github/workflows/agreements.yaml [hunk 1]
```diff
@@ -6,53 +6,8 @@ on:
...
-          remote-repository-name: cla-agreement
-          custom-pr-sign-comment: "I have read the Code of Conduct and I hereby accept the Terms"
-          create-file-commit-message: "For example: Creating file for storing COC Signatures"
-          signed-commit-message: "$contributorName has signed the COC in #$pullRequestNo"
-          custom-notsigned-prcomment: "All contributors have NOT signed the COC Document"
-          custom-allsigned-prcomment: "****CLA Assistant Lite bot**** All contributors have signed the COC  ✍️ ✅"
+  call-workflow-agreements:
+    uses: splunk/addonfactory-github-workflows/.github/workflows/reusable-agreements.yaml@v1.0.0
+    secrets:
+      GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
```
Comment (uoboda-splunk) [inline enforce/.github/workflows/agreements.yaml:12]: Did you test it with any TA repo? I think this token might not work

**Comment by uoboda-splunk** [comment]
Very nice improvement!👏

File: enforce/.github/workflows/agreements.yaml (hunk 1 above)
Comment (arys-splunk) [reply enforce/.github/workflows/agreements.yaml:12]: I did not, but UCC-related open source libraries have the same secrets as any other add-on, so it should work.

Comment (uoboda-splunk) [reply enforce/.github/workflows/agreements.yaml:12]: You are right. I missed that `splunk/addonfactory-github-workflow` is a public repo

**Review by uoboda-splunk — APPROVED** [review]

**Pull request merged**